### PR TITLE
GH-109190: Copyedit 3.12 What's New: Bytecode

### PR DESCRIPTION
**Pull request by AA-Turner**
* Mostly cribbed from @MatthieuDartiailh's list in #107457, save for not covering 
FOR_ITER's new behaviour; the changes to the POP_JUMP_IF_* family of instructions; and the 
YIELD_VALUE change
* I also noticed that ``sys.monitoring`` wasn't covered in "improved modules", so I've added that as it's somewhat related

* Issue: gh-109190

----
:books: Documentation preview :books:: https://cpython-previews--109821.org.readthedocs.build/en/109821/whatsnew/3.12.html

### CODE REVIEW

File: Doc/whatsnew/3.12.rst [hunk 1]
```diff
@@ -1020,9 +1027,27 @@ CPython bytecode changes
 * Removed the :opcode:`!PRECALL` instruction. (Contributed by Mark Shannon in
   :gh:`92925`.)
 
+* Add the :opcode:`BINARY_SLICE` and :opcode:`STORE_SLICE` instructions.
+  (Contributed by Mark Shannon in :gh:`94163`.)
+
+* Add the :opcode:`CALL_INSTRINSIC_1` and :opcode:`CALL_INSTRINSIC_2` instructions.
+  (Contributed by Mark Shannon in :gh:`99005`.)
+
+* Add the :opcode:`CALL_INSTRINSIC_2` instruction.
```
Comment (MatthieuDartiailh) [inline Doc/whatsnew/3.12.rst]: `CALL_INSTRINSIC_2` is listed twice

File: Doc/whatsnew/3.12.rst [hunk 2]
```diff
@@ -1020,9 +1027,27 @@ CPython bytecode changes
...
+  (Contributed by Mark Shannon in :gh:`99005`.)
+
+* Add the :opcode:`!CALL_INSTRINSIC_2` instruction.
+  (Contributed by Irit Katriel in :gh:`101799`.)
+
+* Add the :opcode:`CLEANUP_THROW` instruction.
+  (Contributed by Brandt Bucher in :gh:`90997`.)
+
+* Add the :opcode:`!END_SEND` instruction.
+  (Contributed by Mark Shannon in :gh:`103082`.)
```
Comment (hugovk) [inline Doc/whatsnew/3.12.rst]: Can we document `CALL_INSTRINSIC_1`, `CALL_INSTRINSIC_2` and `END_SEND` opcodes on the `dis` module page, like the others, and then link to them?

Comment (AA-Turner) [reply Doc/whatsnew/3.12.rst]: This is @MatthieuDartiailh's PR (#108900). ~~Though CALL_INTRINSIC_1/2 seem to already be documented, unsure why these weren't being picked up.~~

A

**Review by hugovk — APPROVED** [review]
Nice catch on the `INTRINSIC` typo :)

Let's merge, we can update later if `END_SEND` is ever documented (via https://github.com/python/cpython/issues/107457).

File: Doc/whatsnew/3.12.rst [hunk 3]
```diff
@@ -1032,6 +1057,8 @@ CPython bytecode changes
 * Add the :opcode:`LOAD_SUPER_ATTR` instruction. (Contributed by Carl Meyer and
   Vladimir Matveev in :gh:`103497`.)
 
+* Add the :opcode:`RETURN_CONST` instruction. (:gh:`101632`.)
```
Comment (hugovk) [inline Doc/whatsnew/3.12.rst]: Add credit for issue https://github.com/python/cpython/issues/101632 / PR https://github.com/python/cpython/pull/101633?

```suggestion
* Add the :opcode:`RETURN_CONST` instruction. (Contributed by penguin_wwy in :gh:`101632`.)
```

Comment (AA-Turner) [reply Doc/whatsnew/3.12.rst]: @penguin-wwy please could we check how you'd like to be credited for this change? (I have a slight preference against using usernames in the documentation!)

A

Comment (penguin-wwy) [reply Doc/whatsnew/3.12.rst]: Tanks, if possible, then just use Wenyang Wang.

File: Doc/whatsnew/3.12.rst [hunk 4]
```diff
@@ -1095,12 +1120,7 @@ CPython bytecode changes
 * Add the :opcode:`LOAD_SUPER_ATTR` instruction. (Contributed by Carl Meyer and
   Vladimir Matveev in :gh:`103497`.)
 
-FOR_ITER new behavior is not mentioned
-The fact that POP_JUMP_IF_* family of instructions are now real instructions is not mentioned
-YIELD_VALUE need for an argument is not mentioned
-
-
-
+* Add the :opcode:`RETURN_CONST` instruction. (:gh:`101632`.)
```
Comment (hugovk) [inline Doc/whatsnew/3.12.rst]:
```suggestion
* Add the :opcode:`RETURN_CONST` instruction. (Contributed by Wenyang Wang in :gh:`101632`.)
```

File: Doc/whatsnew/3.12.rst (hunk 3 above)
Comment (hugovk) [reply Doc/whatsnew/3.12.rst]: Thank you!

**Pull request merged**